### PR TITLE
ci: Update workflows based on http-proxy's

### PR DESCRIPTION
**Pull request by Gelbpunkt**
This is technically a breaking change because we used to push images to Docker Hub (undocumented). This PR replaces it with the GitHub Container Registry. It's quite a hassle to push to both registries and Docker Hub's recent changes to announce removing free organizations and paddling backwards a few days later makes it the option with a less certain future. I want to rehaul the README entirely in a different PR.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          components: clippy
+
+      - name: Setup cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-clippy-rustc-${{ steps.toolchain.outputs.cachekey }}-${ hashfiles('**/Cargo.lock') }}
```
Comment (vilgotf) [inline .github/workflows/ci.yml]: Pleace replace this (and all other instances) with the more optimized https://github.com/Swatinem/rust-cache action

**Review by vilgotf — CHANGES_REQUESTED** [review]
Publishing to ghcr instead of dockerhub is fine as the old version is still available but perhaps we should mention this in an annoucement in the Discord server for all to see? Perhaps also in this repo's README?

**Comment by Gelbpunkt** [comment]
> Publishing to ghcr instead of dockerhub is fine as the old version is still available but perhaps we should mention this in an annoucement in the Discord server for all to see? Perhaps also in this repo's README?

I updated the README

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+name: CI
+
+on:
+  workflow_dispatch:
```
Comment (vilgotf) [inline .github/workflows/ci.yml:4]: typo? why run CI on workflow_dispatch?
```suggestion
```

**Review by vilgotf — APPROVED** [review]
LGTM, but I'm sure we can improve it even further down the line.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Gelbpunkt) [reply .github/workflows/ci.yml:4]: Why would we remove this? It might be useful in the future

**Pull request merged**